pyfr/plugins/tavg.py

```diff
@@ -351,3 +357,200 @@ def __call__(self, intg):
 
                 self.tout_last = intg.tcurr
+
+
+class TavgCLIPlugin(BaseCLIPlugin):
+    name = 'tavg'
+    systems = ['*']
+    formulations = ['dual', 'std']
+
+    @classmethod
+    def parse_cli(cls, parser):
+        cmd = cls.name
+        cp = parser.add_parser(cmd, help=f'{cmd} --help')
+        sp = cp.add_subparsers(dest='sub_cmd')
+
+        # Merge command
+        scmd = 'merge'
+        ssp = sp.add_parser(scmd, help=f'{cmd} {scmd} --help')
+        ssp.set_defaults(process=TavgCLIPlugin.merge_cli)
+        ssp.add_argument('solns', nargs='*', help='Averages to merge')
+        ssp.add_argument('output', help='Output file name')
+
+    @classmethod
+    def merge_cli(cls, args):
+        merge = cls()
+        merge._merge_cli(args)
+
+    def _merge_cli(self, args):
+        self.args = args
+
+        # Open all the solition files
+        self._prepare_files(args.solns)
+
+        # Build up which files are needed and if std can be created
+        active = self._active_files()
+        merged_times = [(r.start, r.end) for r, f, s in active]
+        avg_time = sum([t[1] - t[0] for t in merged_times])
+        self.std_all = all([s == 'all' for r, f, s in active])
+
+        # Initialise things needed for the merge
+        r0, f0, s0 = active[0]
+        self._init_tavg_merge(f0)
+
+        # Check all files are compatible for merge
+        self._compatibility_merge(active)
+
+        with h5py.File(args.output, 'w') as out:
+            # Merge the averages and stds
+            self._merge_data(out, active, avg_time)
+
+            # Merge runtime stats
+            self._merge_runtime_stats(active)
+
+            # Merge the metadata
+            self._merge_meta(out, merged_times)
+
+    def _active_files(self):
+        times = []
+        for files in self.sfiles.values():
+            file, stat, cfg = files
+            cfgsect = stat.get('tavg', 'cfg-section')
+            std = cfg.get(cfgsect, 'std-mode')
+            twindow = stat.getliteral('tavg', 'range')
+
+            for w in twindow:
+                times.append((Interval(*w), files, std))
+
+        active = times
+        for i, (r, f, s) in enumerate(times):
+            if any([(r in R and f != F) for R, F, S in active]):
+                active.pop(i)
+            elif any([r.o_intersection(R) for R, F, S in active]):
+                raise ValueError('Clash of average file time windows')
+        return active
+
+    def _compatibility_merge(self, active):
+        for t_rng, (file, stat, cfg), std_mode in active:
+            cfgsect = stat.get('tavg', 'cfg-section')
+            fields = stat.get('data', 'fields').split(',')
+
+            if self.uuid != file['mesh_uuid']:
+                raise RuntimeError('Average files computed on different meshes')
+            if self.region != cfg.get(cfgsect, 'region'):
+                raise RuntimeError('Average files computed on different regions')
+            if self.fields != fields:
```

Comment:
No need to check this (if you verify the average fields match up you're good). Also enables functional expressions to change (which is something we should allow).  Although do allow that this does mean indexes can change, although this should be easy enough to deal with.